src/lib/errors.ts

```diff
@@ -0,0 +1,8 @@
+class IntegrationError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = 'INTEGRATION ERROR'
+  }
+}
+
+export const missingProviderError = new IntegrationError('Missing provider')
```

Comment:
Error definitions that I'm used to (including native errors) just export classes, and have you check equality with `instanceof`. Can this error instead extend an integration error, and just be exported as a class?

```
export class MissingProviderError extends IntegrationError {
  constructor() {
    super('Missing provider')
  }
}
```

This would change its usage in the generator to `throw new MissingProviderError()`.